marconi-chain-index/src/Marconi/ChainIndex/CLI.hs

```diff
@@ -42,38 +42,84 @@ import Marconi.ChainIndex.Types (
 
-{- | Allow the user to set a starting point for indexing the user needs to provide both
- a @BlockHeaderHash@ (encoded in RawBytesHex) and a @SlotNo@ (a natural number).
+-- | Represents a specified point from which to start indexing
+data StartingPoint
+  = StartFromGenesis
+  | StartFromLastSyncPoint
+  | StartFrom ChainPoint
+  deriving (Show, Eq, Generic, ToJSON, FromJSON)
+
+{- | Allow the user to set a starting point for indexing.
+
+  * If the user picks a specific point with the @start-from@ command, they need to provide both
+      a @BlockHeaderHash@ (encoded in RawBytesHex) and a @SlotNo@ (a natural number).
+  * If the chooses either @--start-from-genesis@ or @--start-from-last-sync-points@, no args
+      are required.
+  * Not using any of these commands results in the same behaviour as @--start-from-last-sync-points@
 -}
-chainPointParser :: Opt.Parser C.ChainPoint
-chainPointParser =
-  pure C.ChainPointAtGenesis Opt.<|> (C.ChainPoint <$> slotNoParser <*> blockHeaderHashParser)
+commonStartFromParser :: Opt.Parser StartingPoint
+commonStartFromParser =
+  pure StartFromLastSyncPoint
+    Opt.<|> fromGenesis
+    Opt.<|> fromLastSyncPoint
+    Opt.<|> givenPoint
   where
-    blockHeaderHashParser :: Opt.Parser (C.Hash C.BlockHeader)
-    blockHeaderHashParser =
-      Opt.option
-        (Opt.maybeReader maybeParseHashBlockHeader Opt.<|> Opt.readerError "Malformed block header hash")
-        ( Opt.long "block-header-hash"
-            <> Opt.short 'b'
-            <> Opt.metavar "BLOCK-HEADER-HASH"
-            <> Opt.help
-              "Block header hash of the preferred starting point. Note that you also need to provide the starting point slot number with `--slot-no`. Might fail if the target indexers can't resume from arbitrary points."
+    fromGenesis =
+      Opt.flag'
+        StartFromGenesis
+        ( Opt.long "start-from-genesis"
+            <> Opt.help "Start from genesis"
         )
-    slotNoParser :: Opt.Parser C.SlotNo
-    slotNoParser =
-      Opt.option
-        (C.SlotNo <$> Opt.auto)
-        ( Opt.long "slot-no"
-            <> Opt.short 'n'
-            <> Opt.metavar "SLOT-NO"
-            <> Opt.help
-              "Slot number of the preferred starting point. Note that you also need to provide the starting point block header hash with `--block-header-hash`. Might fail if the target indexers can't resume from arbitrary points."
+    fromLastSyncPoint =
+      Opt.flag'
+        StartFromLastSyncPoint
+        ( Opt.long "start-from-last-sync-points"
+            <> Opt.help "Start from the minimum last sync point"
         )
-    maybeParseHashBlockHeader :: String -> Maybe (C.Hash C.BlockHeader)
-    maybeParseHashBlockHeader =
-      either (const Nothing) Just
-        . C.deserialiseFromRawBytesHex (C.proxyToAsType Proxy)
-        . C8.pack
+    givenPoint :: Opt.Parser StartingPoint
+    givenPoint =
+      StartFrom
+        <$> Opt.hsubparser
+          ( Opt.command
```

Comment:
But why a command? I feel like you could omit the `start-from`, and have instead `--start-from (--slotNo ... --block-header-hash ...)`. Unless optparse doesn't permit that.

Reply:
We can, but it lets you do weird stuff like this (not that anyone actually would, hopefully):

`--start-from --node-config-path ~/git/marconi/config/cardano-node/preprod/config.json --slot-no 1 --block-header-hash asdasd`

Reply:
`--start-from slotNo:blockHeaderHash`? Having slot number alone and not being able to process it is misleading anyway.